src/Validator/Rules/CustomValidationRule.php

```diff
@@ -10,7 +10,7 @@
 class CustomValidationRule extends ValidationRule
 {
     /** @var callable */
-    private $visitorFn;
+    protected $visitorFn;
```

Comment:
```suggestion
    protected callable $visitorFn;
```

Same with other properties, let's do it now to save you the inevitable breakage later ;)

Reply:
It is not possible to use callable to type properties because it is [inconsistent](https://wiki.php.net/rfc/consistent_callables). AI'm not a 100% sure of this for php 8.0 but in 7.4 this not work for me, in my projects.